Permissions in admin templates

reinaris opened the pull request:
I implemented basic permissions in the admin templates.

reinaris committed:
Implemented permissions in product admin templates

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://reinaris-admin-product-abilities.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://reinaris-admin-product-abilities.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

JDutil commented on core/app/models/spree/ability.rb:
```diff
@@ -32,7 +32,7 @@ def initialize(user)
       alias_action :show, to: :read
       alias_action :index, :read, to: :display
-
+      alias_action :create, :update, :destroy, to: :modify
```
Why not just use :manage ?

reinaris replied:
@JDutil I can't discover what :manage stands for, is it exactly the same (create, update, destroy)? In that case we should :)

simmsy replied:
@JDutil manage implies you can do all actions on an object. For example you may be able to create, edit or destroy an order (therefore view it's edit screen) but you cannot approve an order (which would be included with the manage action). modify implies you can mutates an object BUT not necessarily perform all actions.

JDutil commented:
Would like other peoples thoughts on the alias names.  assign might not be explicit enough I'm not sure, but conceptually I'm alright with the change.

simmsy commented:
@JDutil other more programming friendly term for assign would be relate?

peterberkenbosch commented:
Personally I think assign is the correct wording here. That's exactly what an user is allowed or disallowed to do right? :+1:

reinaris committed:
Abilitys in order pages

reinaris committed:
Added basic cancan to all configuration templates

reinaris commented:
I made this PR more general for all admin templates.

I deleted the new alias from this PR to make it more easy.
Will open a new PR for this, so we can discus there.

reinaris committed:
Changed assign to edit in product form

Deleted the new assign alias to make this PR only about the permissions
in general

athal7 commented:
:+1:

peterberkenbosch commented:
Like it!, just 2 failing specs left @reinaris. Could you take look?

```
1) Tax Categories admin creating new tax category should be able to create new tax category
     Failure/Error: Unable to find matching line from backtrace
     ActionView::Template::Error:
       undefined local variable or method `tax_category' for #<#<Class:0x0000000fd99e50>:0x0000001177e230>
     # /home/ubuntu/spree/core/app/helpers/spree/base_helper.rb:68:in `method_missing'
     # ./app/views/spree/admin/tax_categories/index.html.erb:40:in `_1c744aa660a9e8aa540acb51a67856b2'

  2) Tax Categories admin creating new tax category should show validation errors if there are any
     Failure/Error: Unable to find matching line from backtrace
     ActionView::Template::Error:
       undefined local variable or method `tax_category' for #<#<Class:0x0000000fd99e50>:0x00000010a6d208>
     # /home/ubuntu/spree/core/app/helpers/spree/base_helper.rb:68:in `method_missing'
     # ./app/views/spree/admin/tax_categories/index.html.erb:40:in `_1c744aa660a9e8aa540acb51a67856b2'
```

reinaris commented:
@peterberkenbosch I will have a look tomorrow!

reinaris commented:
@peterberkenbosch locally they just pass. From the backend app I run ` bundle exec rspec spec/features/admin/configuration/tax_categories_spec.rb`.
No errors.

reinaris committed:
Used wrong object in cancan which caused error

reinaris commented:
@peterberkenbosch fixed it!

reinaris commented:
New errors showed up, it this because I rebased against master, because I didn't changed anything else then those tax_category fix.
